### DISCUSSION TITLE
LBB Won't Do Anything

### DISCUSSION THREAD

**Discussion by ghost** [outdoorbits/little-backup-box#154] [Q&A]
So about 3 months ago I setup LBB on a raspberry pi 4 so that I wouldn't have to lug my laptop around in the field. Tested it out numerous times using my two Anker power banks. Backed up cards to my hard drives no issue everytime. Now I am blown away by the first time I'm in the field relying on it, it's shitting the bed. I plug it into the power source and open up the web interface and the first thing I noticed is the log is showing nothing other than "Little Backup Box" where it normally shows more on startup. I plug in my hard drive, does nothing, I plug in my card reader does nothing. I can go and manually mount USB 1 and 2 but then when I click source to ext backup like it's supposed to automatically do when plugging those two things in all it does it right away unmount both. 

I am also unable to even power off or reboot LBB from the interface. It just hangs and does nothing. I have literally changed nothing with this just been using it as described. What is going on? I have limited resources as I'm in the mountains of Pakistan right. I'm a bit frustrated since it has never done this and worked flawlessly until I needed it. Any and all help greatly appreciated.

**Comment by ghost** [comment]
I'd like to add that I have it set to maximum logging and it shows close to nothing. If there is even a way to go in and manually transfer the files I'd take that right now. I'm in the field for two weeks and while I have a number of memory cards there's no way I can survive off of just those. I planned for being able to backup cards and clear them.

**Reply by outdoorbits** [reply]
Just to narrow down the stage of development: How old is the installation exactly?
Mounting devices before backup is irrelevant as the backup process at first removes all the mounts.
Please try another way: Unplug your USB devices and start your backup. On the display as like on the log you get instructions, what to do. Just follow them. Do you backup directly from camera or from cardreader only? If you backup from camera you most probably have to switch on your camera before your connect it.

**Reply by outdoorbits** [reply]
Example: Backup from cardreader to external storage:
- Boot LBB and click on storage to external.
- you'll get the instruction to connect your external storage (target device).
- you'll get some information about this storage
- next instruction is to connect your source device
- backup should start, you get progress info on display as on the log

Can you reproduce this?

**Reply by ghost** [reply]
Hey thanks for the reply. For whatever reason it just started working. I'll try to explain what I did (not sure if it is what made it start working) and I have the log file attached).

To answer your questions first, the installation is only 3-4 months old. So sometime in March or April. I am backing up from a card reader to a portable hard drive.

After force restarting it (unplugging it) several times I was looking for a way to copy the files manually. After hitting "mount USB 1" and likewise for USB 2 I could browse both devices in the file manager in the interface. I initiated a copy of the entire sd card to the hard drive. Since it doesn't really give you any information on the status of the copy operation in the file manager I was just gonna let it sit for an hour and browse the devices to see if everything copied. Came back 15 minutes later and pulled up the LBB web interface and the log showed it was in the middle of a source to ext backup operation. I let it finish. Was able to shut it down via the interface and then I went ahead and plugged in my second hard drive and then sd card and it started automatically backing up. 

It was about a 2 hours the whole thing wasn't working as is expected. It was exactly as described by Ritzelschrubber in the "No reaction after startup" discussion.

Hope the log helps in pinning down why it happened. 
[little-backup-box.log](https://github.com/outdoorbits/little-backup-box/files/8942623/little-backup-box.log)

**Reply by outdoorbits** [reply]
1. Plugging off the power while the pi is up (green led working) can destroy your installation. Don't do this! Please!

2. If you connect source and target before getting the instruction to connect, the direction of the backup just random!

3. If there are any problems, use tools and checkdisk. Your storage was unplugged unclean.

**Reply by ghost** [reply]
If I run into this again I'll try what you mentioned before. Hopefully I don't. 

Yes I know that just unplugging the power on a device is a mortal sin and there is a 10th circle of hell for people who do such things but I had no other way to shut it off! Anyway, thanks for the prompt replies.

**Reply by outdoorbits** [reply]
Please check your setup, there is an option for idle time to shut down if no backup is running and there is no web access. Could help in such situations.

**Comment by ghost** [comment]
Just wanted to add some details to this in case anyone runs into this issue in the future. 

Despite what I mentioned earlier about abruptly disconnecting LBB from its power source to shut it down while it wasn't working, this is not something I do normally at all. 

The other day someone decided they needed to use the power bank powering my LBB setup and proceeded to abruptly disconnect it. Luckily file transfers had already finished so there was no corruption. However, when I powered it up again, I was having this same issue described initially in this thread. So I went and set the auto-shutdown timer to 2 minutes and let it gracefully shut down this time. Powered it up again and everything was fine.

So these issues I was experiencing appear to stem from an un-graceful shutdown. I'm not sure how that happened the first time around since I don't just unplug the device. 

So from now on, I'm going to leave the auto-shutdown timer on just in case of well anything. And as double protection against complete failure in the field, I'm going to start carrying a second microSD card loaded with LBB just in case.
